enrichment-definition/src/main/java/eu/europeana/enrichment/model/impl/TopicEntityImpl.java

```diff
@@ -0,0 +1,219 @@
+package eu.europeana.enrichment.model.impl;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.bson.types.ObjectId;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import dev.morphia.annotations.Entity;
+import dev.morphia.annotations.Id;
+import dev.morphia.annotations.Transient;
+import eu.europeana.enrichment.model.TermEntity;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.model.TopicModel;
+
+
+@Entity(value="TopicEntityImpl")
+public class TopicEntityImpl implements TopicEntity {
+	
+	//id will be used for storing MongoDB _id
+	@Id
+	@JsonIgnore
+	public String _id = new ObjectId().toString();
+	
+	@Transient
+	private TopicModel model;
```

Comment:
We can only keep the id of the topic model, no need to have the whole TopicModel object.